screen-main/src/main/java/com/ivy/main/MainBottomBar.kt

```diff
@@ -180,23 +184,41 @@ fun BoxWithConstraintsScope.BottomBar(
     var dragOffset by remember {
         mutableStateOf(Offset.Zero)
     }
-    // + & x button
+
+    val fabStartXOffset = FAB_BUTTON_SIZE.toDensityPx()
+    val fabStartYOffset = navigationBarInset() + FAB_BUTTON_SIZE.toDensityPx()
+    val addIvyCircleModifier = Modifier
+//    // + & x button
     IvyCircleButton(
-        modifier = Modifier
+        modifier = addIvyCircleModifier
             .layout { measurable, constraints ->
-                val placeable = measurable.measure(constraints)
+
+                val maxWidthAllowedByParent = constraints.maxWidth
+                val maxHeightAllowedByParent = constraints.maxHeight
+                Timber.d("maxHeightAllowedByParent $maxHeightAllowedByParent")
+                Timber.d("maxWidthAllowedByParent $maxWidthAllowedByParent")
+
+                val placeable = measurable.measure(
+                    constraints = constraints
+                )
                 layout(placeable.width, placeable.height) {
                     placeable.place(
-                        x = fabStartX.roundToInt(),
-                        y = fabStartY.roundToInt()
+//                        x = fabStartX.roundToInt(),
+//                        y = fabStartY.roundToInt()
+
+                        x = (maxWidthAllowedByParent / 2 - fabStartXOffset / 2).roundToInt(),
+                        y = (maxHeightAllowedByParent - fabStartYOffset).roundToInt()
+
                     )
                 }
             }
             .size(FAB_BUTTON_SIZE)
             .rotate(fabRotation)
             .zIndex(200f)
             .thenIf(tab == MainTab.HOME) {
-                pointerInput(Unit) {
+                /** Here We are getting some other instance of modifier not the one we are working with
+                thats why i saved the reference of the modifier we were working with and used its pointerInput() **/
+                addIvyCircleModifier.pointerInput(Unit) {
```

Comment:
Seems like that the `thenIf` context receivers "this" isn't the correct modifier. If you fix the `thenIf` to pass.`addIvyCircleModifier` as **this** (context receiver) it should work